provider/k8s/metric_scraper.go

```diff
@@ -0,0 +1,154 @@
+package k8s
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/convox/convox/pkg/structs"
+	"github.com/pkg/errors"
+	am "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type MetricScraperClient struct {
+	c       *http.Client
+	cluster kubernetes.Interface
+	host    string
+}
+
+func NewMetricScraperClient(cluster kubernetes.Interface, host string) *MetricScraperClient {
+	return &MetricScraperClient{
+		c: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+		cluster: cluster,
+		host:    host,
+	}
+}
+
+func (m *MetricScraperClient) GetRackMetrics(opts structs.MetricsOptions) (structs.Metrics, error) {
+	if m.host == "" {
+		return nil, errors.WithStack(fmt.Errorf("unimplemented"))
+	}
+
+	ns, err := m.cluster.CoreV1().Nodes().List(am.ListOptions{})
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	nodeNames := []string{}
+	var cpuAllocatable, memAllocatable float64
+	for _, n := range ns.Items {
+		nodeNames = append(nodeNames, n.ObjectMeta.Name)
+		cpuAllocatable += float64(n.Status.Allocatable.Cpu().MilliValue())
+		memAllocatable += float64(n.Status.Allocatable.Memory().Value())
+	}
+
+	cpus, err := m.GetNodesMetrics(strings.Join(nodeNames, ","), structs.ScraperMetricTypeCpu)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	if len(cpus.Items) == 0 {
+		return nil, fmt.Errorf("metrics unavailable")
+	}
```

Comment:
Return a message with more context (?). Something like:

```suggestion
	if len(cpus.Items) == 0 {
		return nil, fmt.Errorf("metrics unavailable, CPU units are nil")
	}
```